addons/account/static/tests/reconciliation_tests.js

```diff
@@ -1620,5 +1622,135 @@ QUnit.module('account', {
+                        case 'account.tax':
+                            assert.ok(true, "Taxes list has been fetched");
+                            return $.when(
+                                _.map(this.data[args.model].records, function (record) {
+                                    return [record.id, record.display_name];
+                                })
+                            );
+                        case 'account.journal':
+                            assert.ok(true, "Journals list has been fetched");
+                            return $.when(
+                                _.map(this.data[args.model].records, function (record) {
+                                    return [record.id, record.display_name];
+                                })
+                            );
+                    }
+                }
+                if (args.method === 'process_move_lines') {
+                    var mv_line_ids = args.args[0][0].mv_line_ids.slice(0);
+                    console.log(mv_line_ids)
+                    mv_line_ids.sort(function(a, b) {return a - b});
+
+                    switch (reconcileNo) {
+                        case 1:
+                            assert.deepEqual(mv_line_ids, [399, 402], "1st reconciliation: Lines to be moved are the correct ones");
+                            break
+                        case 2:
+                            assert.deepEqual(mv_line_ids, [6, 19, 21], "2nd reconciliation: Lines to be moved are the correct ones");
+
+                            var idx = _.has(args.args[0][0].new_mv_line_dicts[0], 'journal_id') ? 0 : 1; // Index aiming at the correct object in the list
+                            assert.deepEqual(
+                                _.pick(args.args[0][0].new_mv_line_dicts[idx],
+                                       'account_id', 'name', 'credit', 'debit', 'journal_id'),
+                                {account_id: 287, name: "dummy text", credit: 0, debit: 180, journal_id: 8},
+                                "2nd reconciliation: The gift line is correct"
+                            );
+                            assert.deepEqual(
+                                _.pick(args.args[0][0].new_mv_line_dicts[1 - idx],
+                                       'account_id', 'name', 'credit', 'debit', 'tax_line_id'),
+                                {account_id: 287, name: "Tax 20.00%", credit: 0, debit: 36, tax_line_id: 6},
+                                "2nd reconciliation: The tax line is correct"
+                            );
+                            break
+                    }
+                }
+                return this._super.apply(this, arguments);
+            }
+        });
+
+        var $body = $('body');
+        $body.addClass('debug');
+        clientAction.appendTo($body);
+
+        // First account is already in the wanted state to test the
+        // reconciliation. We reconcile it now.
+        reconcileNo = 1;
```

Comment:
well....

Reply:
it's the easiest way to run the two tests

Reply:
Maybe I'm missing something but I'm thinking this is not needed.
It seems to me that all the stuff with the reconcileNo and the mockup above are there only to remove the first account that is already ready to be reconciled and then the actual test is on the second account once the first one had disappeared. Why not directly test on the second one and skip the reconciliation of the first one which does not seems relevant to the issue.

Change the mockup and remove the switch, only leave the part that was for the reconcileNo == 2
Remove the two lines
reconcileNo = 1 and
clientAction.$("button.o_reconcile.btn.btn-primary:first").click();

Instead: take the second widget:
var widget = clientAction.widgets[1];
trigger a click on it to open it:
clientAction.widgets[1].$('.accounting_view thead td:first').trigger('click'); (or something like that, need to check, this is from another test above in the file)
and execute all your operation on this widget:
so for example:
widget.$('tr.mv_line[]')
and so on.